src/Document/Obstacles/Material.ts

```diff
@@ -0,0 +1,66 @@
+import { Vector4F } from '../../Utilities/types';
+import { INameable } from '../Attributes/INameable';
+import { bzwBool, bzwFloat, bzwString, bzwVector4F } from '../attributeParsers';
+import { IBaseObject, newIBaseObject } from './BaseObject';
+
+export const MaterialProperties = {
+  name: bzwString,
+  texture: bzwString,
+  addtexture: bzwString,
+  notextures: bzwBool,
+  notexcolor: bzwBool,
+  notexalpha: bzwBool,
+  texmat: bzwString,
+  dyncol: bzwString,
+  ambient: bzwVector4F,
+  diffuse: bzwVector4F,
+  color: bzwVector4F,
+  specular: bzwVector4F,
+  emission: bzwVector4F,
+  shininess: bzwFloat,
+  resetmat: bzwBool,
+  spheremap: bzwBool,
+  noradar: bzwBool,
+  noshadow: bzwBool,
+  noculling: bzwBool,
+  nosorting: bzwBool,
+  nolighting: bzwBool,
+  alphathresh: bzwFloat,
+  groupalpha: bzwBool,
+  occluder: bzwBool,
+};
+
+export interface IMaterial extends IBaseObject, INameable {
+  texture: String;
+  addtexture?: String;
+  notextures?: boolean;
+  notexcolor?: boolean;
+  notexalpha?: boolean;
+  texmat: String;
+  dyncol: String;
+  ambient?: Vector4F;
+  diffuse?: Vector4F;
+  color?: Vector4F;
+  specular?: Vector4F;
+  emission?: Vector4F;
+  shininess?: Number;
+  resetmat?: boolean;
+  spheremap?: boolean;
+  noradar?: boolean;
+  noshadow?: boolean;
+  noculling?: boolean;
+  nosorting?: boolean;
+  nolighting?: boolean;
+  alphathresh?: Number;
+  groupalpha?: boolean;
+  occluder?: boolean;
```

Comment:
```suggestion
  texture?: string;
  addtexture?: string;
  notextures?: boolean;
  notexcolor?: boolean;
  notexalpha?: boolean;
  texmat: string;
  dyncol: string;
  ambient?: Vector4F;
  diffuse?: Vector4F;
  color?: Vector4F;
  specular?: Vector4F;
  emission?: Vector4F;
  shininess?: number;
  resetmat?: boolean;
  spheremap?: boolean;
  noradar?: boolean;
  noshadow?: boolean;
  noculling?: boolean;
  nosorting?: boolean;
  nolighting?: boolean;
  alphathresh?: number;
  groupalpha?: boolean;
  occluder?: boolean;
```

While [primitives](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#the-primitives-string-number-and-boolean) are case insensitive, the convention is to have primitives lowercase. Custom types and classes are TitleCase.

Also, see my comment below about `texture` but let's make it optional.

Reply:
Thanks for the info about primitives. I didn't realize that the uppercase versions of JavaScript primitives in TypeScript "refer to some special built-in types". I'm just used to using the uppercase in plain JavaScript (by habits, I should have all the Booleans uppercased as well).